kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue

```diff
@@ -245,6 +281,75 @@
         const newMsg = createReturnMsg({ message, data: __version });
         return this.hashi.mediator.sendMessage(newMsg);
       },
+      sendChannelMetadata(message) {
+        return ChannelResource.fetchModel({ id: this.topic.channel_id })
+          .then(channel => {
+            return createReturnMsg({
+              message,
+              data: {
+                id: channel.id,
+                name: channel.name,
+                description: channel.description,
+                thumbnail: channel.thumbnail,
+              },
+            });
+          })
+          .catch(err => {
+            return createReturnMsg({ message, err });
+          })
+          .then(newMsg => {
+            this.hashi.mediator.sendMessage(newMsg);
+          });
+      },
+      sendChannelFilterOptions(message) {
+        return ChannelResource.fetchFilterOptions(this.topic.channel_id)
+          .then(response => {
+            return createReturnMsg({
+              message,
+              data: {
+                availableAuthors: response.data.available_authors,
+                availableTags: response.data.available_tags,
+                availableKinds: response.data.available_kinds,
+              },
+            });
+          })
+          .catch(err => {
+            return createReturnMsg({ message, err });
+          })
+          .then(newMsg => {
+            this.hashi.mediator.sendMessage(newMsg);
+          });
+      },
+      sendRandomCollection(message) {
+        const { options } = message;
+        const { kinds, onlyContent } = options;
+
+        return ContentNodeResource.fetchRandomCollection({
+          getParams: {
+            parent: options.parent === 'self' ? this.topic.id : options.parent,
+            channel_id: this.topic.channel_id,
+            max_results: options.maxResults ? options.maxResults : 10,
```

Comment:
Pagination forces an ordering, so I am guessing this is not working.

Reply:
This new API method doesn't use the cursor pagination, it doesn't have any pagination, the `max_results` is only used to limit the number of nodes, but it's not possible to get more with the same "random" order, each call should provide a new set of randoms.

https://github.com/learningequality/kolibri/pull/8476/commits/6225201694aa59aa129bd9f97f4acb06656dfcc7#diff-f2b97d70d258e7c248156962c7e63fafe868abced1c0644b8b25a0aa1d0addd2